tests/scripts/create_subset_ebas.py

```diff
@@ -0,0 +1,288 @@
+                merged = sdl.merge_other(sdr, var)
+                if var in merged.overlap:
+                    files[sdl.station_name] = [f0, f1]
+                    overlap_found = True
+    return files, overlap_found
+
+
+def get_files_var_statnum(data, var, statnum):
+    files = {}
+    numfound = 0
+    allstats = data.unique_station_names
+    for stat in allstats:
+        if numfound == statnum:
+            break
+        meta_match = data.find_station_meta_indices(stat)
+
+        for i, idx in enumerate(meta_match):
+            try:
+                sd = data.to_station_data(idx, var)
+            except pya.exceptions.DataCoverageError:
+                continue
+            if len(sd[var].dropna()) > 10:
+                fname = sd.filename
+                if ";" in fname:
+                    raise ValueError("FATAL", var, stat)
+                files[sd.station_name] = [sd.filename]
+                numfound += 1
+                break
+    return files
+
+
+def main():
+
+    # reader = pya.io.ReadUngridded(NAME, data_dir=EBAS_BASE_DIR)
+    reader = pya.io.ReadUngridded(
+        NAME,
+    )
+    r_lowlev = reader.get_lowlevel_reader(NAME)
+
+    # r_lowlev._dataset_path = ebas_local
+
+    # directory containing NASA Ames files
+    FILEDIR_SRC = r_lowlev.file_dir
+    print(f"checking for invalid files in static file list...")
+    CURRENT_OK, files_valid, files_invalid = check_outdated(FILEDIR_SRC)
+    print(f"done...")
+
+    if not CURRENT_OK:
+        print(f"outdated files: {files_invalid}")
+        print("---------------------WARNING------------------------------")
+        print(
+            "Some files in current EBAS subset do not exist anymore in "
+            "the most recent EBAS database, consider updating them"
+        )
+        print("---------------------WARNING------------------------------")
+        raise Exception
+
+    if not OUTBASE.exists():
+        OUTBASE.mkdir()
+
+    if not FILES_DEST.exists():
+        FILES_DEST.mkdir()
+
+    if (
+        SEARCH_PROBLEM_FILES
+    ):  # use this block to find interesting files, e.g. that throw certain exceptions
+        var = "concpm10"
+        f0 = 1100
+        f1 = f0 + 1000
+
+        reader = pya.io.ReadEbas()
+        reader.read(var, first_file=f0, last_file=f1)
+        for i, file in enumerate(reader.files_failed):
+            try:
+                data = reader.read_file(file, var)
+            except Exception as e:
+                print(i)
+                print(os.path.basename(file))
+                print(e)
+                try:
+                    fd = pya.io.EbasNasaAmesFile(file)
+                    print(fd.station_name)
+                except Exception as e:
+                    print(e)
+
+    else:
+        infofile = JSON_FILE
+        if os.path.exists(infofile):
+            with open(infofile, "r") as f:
+                current_files = simplejson.load(f)
+        else:
+            current_files = {}
```

Comment:
`JSON_FILE` is a Path object, so this section can be written as

``` python
        if JSON_FILE.exists():
            current_files = simplejson.loads(JSON_FILE.read_text())
        else:
            current_files = {}
```

Reply:
I put that into a separate issue #604. A bit out of scope for now.